File: Backend/Financeiro.ConciliacaoWebAPI/JsonWebToken/Controllers/IntervaloController.cs
using System;
using System.Collections.Generic;
using JsonWebToken.Code;
using System.Web.Http;

namespace JsonWebToken.Model
{
  public class IntervaloController : ApiController
  {
    // GET api/<controller>
    public IEnumerable<IntervaloModel> Get(int periodo)
    {
      IntervaloBLL bll = new IntervaloBLL();

      return bll.selectIntervalo(periodo);
    }

    // GET api/<controller>/5
  //  public string Get(int id)
  //  {
  //    return "value";
  //  }

    // POST api/<controller>
    public void Post([FromBody]IntervaloModel intervalo)
    {
      IntervaloBLL bll = new IntervaloBLL();
      bll.insertIntervalo(intervalo);
    }

    // PUT api/<controller>/5
    public void Put( [FromBody]IntervaloModel intervalo,string acao)
    {
      IntervaloBLL bll = new IntervaloBLL();
      if (acao.ToLower().Equals("ativar") || acao.ToLower().Equals("desativar"))
      {
        bll.ativarDesativarIntervalo(intervalo);
      }
      else
      {
        bll.updateIntervalo(intervalo);
      }
    }

    // DELETE api/<controller>/5
    public void Delete(int id)
    {
    }
  }
}

File: Backend/Martins.Conciliacao.API/Controllers/IntervaloController.vb
Imports System.Web.Http
Imports Martins.Conciliacao.Code.Intervalo
Imports Martins.Conciliacao.Model

Namespace Controllers
  Public Class IntervaloController
    Inherits ApiController
    ' GET api/<controller>
    Public Function [Get](periodo As Integer) As IEnumerable(Of IntervaloModel)
      Dim bll As New IntervaloBLL()

      Return bll.selectIntervalo(periodo)
    End Function

    ' GET api/<controller>/5
    '  public string Get(int id)
    '  {
    '    return "value";
    '  }

    ' POST api/<controller>
    Public Sub Post(<FromBody> intervalo As IntervaloModel)
      Dim bll As New IntervaloBLL()
      bll.insertIntervalo(intervalo)
    End Sub

    ' PUT api/<controller>/5
    Public Sub Put(<FromBody> intervalo As IntervaloModel, acao As String)
      Dim bll As New IntervaloBLL()
      If acao.ToLower().Equals("ativar") OrElse acao.ToLower().Equals("desativar") Then
        bll.ativarDesativarIntervalo(intervalo)
      Else
        bll.updateIntervalo(intervalo)
      End If
    End Sub

    ' DELETE api/<controller>/5
    Public Sub Delete(id As Integer)
    End Sub
  End Class
End Namespace
